pkg/artifactory/provider.go

```diff
@@ -25,12 +25,11 @@ func Provider() *schema.Provider {
 				ValidateFunc: validation.IsURLWithHTTPorHTTPS,
 			},
 			"username": {
-				Type:          schema.TypeString,
-				Optional:      true,
-				DefaultFunc:   schema.EnvDefaultFunc("ARTIFACTORY_USERNAME", nil),
-				ConflictsWith: []string{"access_token", "api_key"},
```

Comment:
Does `username ` also conflict with `api_key`? Or just `access_token`?

Reply:
ConflictsWith defines if it conflicts with username. So, by removing the field entirely, we solve this problem.

Reply:
That's not what this line says though. It says "username field conflicts with both access_token field and api_key field". So if access_token field is the one in combination with username causing issue, we should still include api_key in this line.

Reply:
testing with api_key